ExampleMod/Items/Accessories/ExclusiveAccessory.cs

```diff
@@ -0,0 +1,185 @@
+			// An intricacy of vanilla is that it directly swaps the items on right click if the items are the same and just their prefixes differ,
+			// even in vanity slots. For this, FindDifferentEquippedExclusiveAccessory() doesn't find these items
+			// That means, if for whatever reason you have Green equipped, Yellow in a vanity slot, and then right click a Yellow item in your inventory
+			// that has a different prefix than the vanity Yellow, it will swap with the vanity Yellow instead of the equipped Green
+			// Therefore we need to reimplement this behavior by doing the following check
+
+			// Check vanity accessory slots for the same item type equipped and return false (so vanilla handles it)
+			int maxAccessoryIndex = 5 + Main.LocalPlayer.extraAccessorySlots;
+			for (int i = 13; i < 13 + maxAccessoryIndex; i++) {
+				if (Main.LocalPlayer.armor[i].type == item.type) return false;
+			}
+
+			bool canRightClick = false;
+			// Only allow right clicking if there is a different ExclusiveAccessory equipped
+			FindDifferentEquippedExclusiveAccessory((int i, Item accessory) => {
+				canRightClick = true;
+			});
+			// If this hook returns true, the item is consumed (just like crates and boss bags)
+			return canRightClick;
+		}
+
+		public override void RightClick(Player player) {
+			// Here we implement the "swapping" when right clicked to equip this item inplace of another one
+			FindDifferentEquippedExclusiveAccessory((int i, Item accessory) => {
+				Main.LocalPlayer.QuickSpawnClonedItem(accessory);
+				// We need to use i instead of foundItem because we directly want to alter the equipped accessory
+				Main.LocalPlayer.armor[i] = item.Clone();
+			});
+		}
+
+		// We make our own method for compacting the code because we will need to check equipped accessories often
+		// This method also has a delegate as an argument, which allows us to pass entire methods instead of just variables
+		// To understand what an Action delegate is, see here: https://www.tutorialsteacher.com/csharp/csharp-action-delegate
+		protected void FindDifferentEquippedExclusiveAccessory(Action<int, Item> whenFound) {
+			int maxAccessoryIndex = 5 + Main.LocalPlayer.extraAccessorySlots;
+			for (int i = 3; i < 3 + maxAccessoryIndex; i++) {
+				Item accessory = Main.LocalPlayer.armor[i];
+				// IsAir makes sure we don't check for "empty" slots
+				// IsTheSameAs() compares two items and returns true if their types match
+				// "is ExclusiveAccessory" is a way of performing pattern matching
+				// Here, inheritance helps us determine if the given item is indeed one of our ExclusiveAccessory ones
+				if (!accessory.IsAir &&
+					!item.IsTheSameAs(accessory) &&
+					accessory.modItem is ExclusiveAccessory) {
+					// If we find an item that matches these criteria, execute the code inside it
+					// The second argument is just for convenience, technically we don't need it since we can get the item from just i
+					whenFound.Invoke(i, accessory);
```

Comment:
Change this to return i, or alternatively change it to return a value tuple as (i, accessory) that can be deconstructed. It keeps the calling code simpler and removes the nesting level.